Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift

```diff
@@ -306,15 +354,35 @@ open class ChatChannelListVC: _ViewController,
     // MARK: - DataControllerStateDelegate
     
     open func controller(_ controller: DataController, didChangeState state: DataController.State) {
+        if !channelListErrorView.isHidden {
+            hideErrorView()
+        }
+        
+        let shouldHideEmptyView: Bool
+        let isLoading: Bool
+        
         switch state {
-        case .initialized, .localDataFetched:
+        case .initialized, .localDataFetched, .remoteDataFetched:
             if self.controller.channels.isEmpty {
-                loadingIndicator.startAnimating()
+                isLoading = true
+                shouldHideEmptyView = false
             } else {
-                loadingIndicator.stopAnimating()
+                isLoading = false
+                shouldHideEmptyView = true
+            }
+        case .localDataFetchFailed, .remoteDataFetchFailed:
+            shouldHideEmptyView = emptyView.isHidden
+            isLoading = false
+            showErrorView()
+        }
+        
+        emptyView.isHidden = shouldHideEmptyView
+        skeletonListView.isHidden = !isLoading
+        
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
```

Comment:
We should cover this logic with unit tests 👍

Reply:
Added a couple of tests that cover the different states cases.